packages/builder/src/pages/builder/portal/users/users/_components/AddUserModal.svelte

```diff
@@ -69,6 +69,14 @@
     return userData[index].error == null
   }
 
+  // Minimum password length is 12 characters, but Math random will generate at most 11 characters, so add another 5.
+  function generatePassword() {
+    return (
+      Math.random().toString(36).substring(2, 22) +
+      Math.random().toString(36).substring(2, 7)
+    )
```

Comment:
This is a lil' janky, and I don't think `Math.random` is guaranteed to be a strong source of randomness.

Could we instead do something like:

```javascript
crypto.randomBytes(15).toString("base64")
```

Reply:
```javascript
function generatePassword(length) {
    const array = new Uint8Array(length);
    window.crypto.getRandomValues(array);
    return Array.from(array, byte => byte.toString(36).padStart(2, '0')).join('').slice(0, length);
}

const password = generatePassword(12);
```

Seems to be the equivalent way to get strong random values in the browser, but it makes me tremendously uneasy we don't do this password generation on the server.

Reply:
Yeah there's really no reason to do that is there. We can just pass up a flag to request an auto-generated password when we create the user